### DISCUSSION TITLE
customize ripple/click effect on mobile

### DISCUSSION THREAD

**Discussion by Darkwebplayer** [skeletonlabs/skeleton#1555] [Q&A] [answered]
### Current Behavior

i want to customise ripple effect/click effect on items like tabs in mobile view, currently on mobile devices it displays a black transparent  box when clicked on the tab(light theme). i need it to be rounded , and change the color to match my theme, is there any way to achieve this?

![photo_2023-05-24_17-01-08](https://github.com/skeletonlabs/skeleton/assets/45286733/74fc8444-38ce-4184-85fc-f9795796bd78)

### Reproduction / Steps To Reproduce

1. stackblitz playground : https://stackblitz.com/edit/skeletonlabs-repl-cdfjkt?file=src%2Froutes%2F%2Bpage.svelte
2. open in new tab and go to mobile view using dev tools
3. click on tab item to see the ripple/click effect

**Comment by endigo9740** [comment, marked as answer]
@Darkwebplayer This fits better within our Discussions > Q&A section rather than a bug issue, so I've moved this accordingly.

Unfortunately we don't really support ripple effects as a standard feature of Skeleton. That's more of an opinionated design choice of Material Design, which we are not and are not trying to emulate. I might recommend one of the various Material Design UI libraries if this is more of the aesthetic you're after:

- https://sveltematerialui.com/
- https://svelte-materialify.vercel.app/

That aside, we are looking to expand the ability to customize transition animations in our components per this on-going task:
- https://github.com/skeletonlabs/skeleton/issues/1322

This will provide you the ability to custom the transition used - either via a canned Svelte transition, or provide your own custom transition. A ripple like effect might be possible using this method, but I'm not sure.

Per the click state you're seeing when tapping the element - that's native browser functionality. Touch browsers have a visual indicator of where you clicked for accessibility purposes. We keep that on very intentionally given our a11y focus. For webkit-based browsers (mobile safari, mobile chrome) we only customize the color from blue to a neutral grey-ish tone:

```css
html {
	-webkit-tap-highlight-color: rgba(128, 128, 128, 0.5);
}
```

You can learn how this works here:
https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-tap-highlight-color

My advice would be to keep this, but customize the color to something that works well with your theme color values.

**Comment by Darkwebplayer** [comment]
Thank you ❤✨️
